### DISCUSSION TITLE
Modal component: make autofocus optional

### DISCUSSION THREAD

**Discussion by gehrisandro** [filamentphp/filament#12977] [Ideas]
Currently, when opening a modal, the focus gets trapped and the first element gets focused.

Trapping the focus seems a valid approach, but focussing the first element can cause some issue.

For example, we have a modal containing a long text with a focusable element (in our case a link) far down. Autofocus on the link will automatically scroll down to the position of the link and the user can not see the beginning of the text without scrolling up again.

In most cases, especially with forms, autofocus seems a reasonable choice, therefore I would suggest leaving the default to autofocus, and add an option to opt-out.

I see two approaches for this, but I am unsure which one is the right, because I am fairly new to Filament.

Option 1: Add an optional property to opt-out.
```blade
<x-filament::modal :autofocus="false">
   ...
</x-filament::modal>
```

Option 2: Allow overriding the x-trap attribute
```blade
<x-filament::modal x-trap.noscroll.noautofocus="isOpen">
   ...
</x-filament::modal>
```

Option 2 gives more flexibility, as it allows adding other [modifiers](https://alpinejs.dev/plugins/focus#modifiers) or to disable x-trap completely. On the other hand, it exposes some implementation details of the modal.
I would prefer option 1.

@danharrin What do you think? If you are interested in this, I am happy to help with a PR.

**Comment by danharrin** [comment]
Hi, I think the new `:autofocus="false"` attribute would be ideal, thank you.

**Reply by gehrisandro** [reply]
Hi @danharrin 

Thanks for your feedback. I've created a PR: https://github.com/filamentphp/filament/pull/12991
